x-pack/legacy/plugins/apm/public/components/app/Settings/AddSettings/AddSettingFlyout.tsx

```diff
@@ -0,0 +1,344 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiPortal,
+  EuiTitle,
+  EuiHorizontalRule,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyoutFooter,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCallOut
+} from '@elastic/eui';
+import React, { useState } from 'react';
+import { toastNotifications } from 'ui/notify';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { AddSettingFlyoutBody } from './AddSettingFlyoutBody';
+import { Config } from '../SettingsList';
+import { useFetcher } from '../../../../hooks/useFetcher';
+import {
+  loadCMServices,
+  loadCMEnvironments,
+  deleteCMConfiguration,
+  updateCMConfiguration,
+  createCMConfiguration
+} from '../../../../services/rest/apm/settings';
+import { ENVIRONMENT_NOT_DEFINED } from '../../../../../common/environment_filter_values';
+
+interface Props {
+  onClose: () => void;
+  onSubmit: () => void;
+  isOpen: boolean;
+  selectedConfig: Config | null;
+}
+
+export function AddSettingsFlyout({
+  onClose,
+  isOpen,
+  onSubmit,
+  selectedConfig
+}: Props) {
+  if (!isOpen) {
+    return null;
+  }
+
+  const [environment, setEnvironment] = useState<string | undefined>(
+    selectedConfig
+      ? selectedConfig.service.environment || ENVIRONMENT_NOT_DEFINED
+      : undefined
+  );
+  const [serviceName, setServiceName] = useState<string | undefined>(
+    selectedConfig ? selectedConfig.service.name : undefined
+  );
+  const [sampleRate, setSampleRate] = useState<number>(
+    selectedConfig ? parseFloat(selectedConfig.settings.sample_rate) : NaN
+  );
+  const { data: serviceNames = [], status: serviceNamesStatus } = useFetcher<
+    string[]
```

Comment:
You don't need to supply a type to `useFetcher`. It is inferred from the promise return type.